projects/aca-shared/rules/src/app.rules.spec.ts

```diff
@@ -540,6 +540,35 @@ describe('app.evaluators', () => {
     });
   });
 
+  describe('isKnowledgeRetrievalEnabled', () => {
+    it('should call context.appConfig.get with correct parameters', () => {
+      context.appConfig = { get: jasmine.createSpy() } as any;
+
+      app.canDisplayKnowledgeRetrievalButton(context);
+      expect(context.appConfig.get).toHaveBeenCalledWith('plugins.knowledgeRetrievalEnabled', true);
+    });
+
+    it('should return true if get from appConfig returns true', () => {
+      expect(
+        app.canDisplayKnowledgeRetrievalButton({
+          appConfig: {
+            get: () => true
+          }
+        } as any)
+      ).toBeTrue();
+    });
+
+    it('should return false if get from appConfig returns false', () => {
+      expect(
+        app.canDisplayKnowledgeRetrievalButton({
+          appConfig: {
+            get: () => false
+          }
+        } as any)
+      ).toBeFalse();
+    });
+  });
```

Comment:
Maybe another test, `should return false if get from appConfig returns true, but navigation is not personalFiles, sharedFiles, recentFiles or favourites`